lib/methods/import.js

```diff
@@ -6,13 +6,65 @@ const { EJSON } = require('bson');
 module.exports = self => {
   return {
     async import(req, moduleName) {
-      const {
-        docs, attachmentsInfo, exportPath, filePath
-      } = await self.readExportFile(req);
       if (!req.user) {
         throw self.apos.error('forbidden');
       }
 
+      const overrideLocale = self.apos.launder.boolean(req.body.overrideLocale);
+      let exportPath = self.apos.launder.string(req.body.exportPath);
+
+      let docs;
+      let attachmentsInfo;
+
+      if (exportPath) {
+        const filesData = await self.getFilesData(exportPath);
+
+        ({ docs, attachmentsInfo } = filesData);
+      } else {
+        const exportFile = await self.readExportFile(req);
+
+        ({ docs, attachmentsInfo } = exportFile);
+        exportPath = exportFile.exportPath;
+
+        await self.cleanFile(exportFile.filePath);
+      }
+
+      const differentDocsLocale = self.getFirstDifferentLocale(req, docs);
+      const siteHasMultipleLocales = Object.keys(self.apos.i18n.locales).length > 1;
+
+      if (differentDocsLocale) {
+        if (siteHasMultipleLocales && !overrideLocale) {
+          // Display confirmation modal to ask the user if he wants to
+          // continue the import with the current locale overriding the docs one:
+          await self.apos.notify(req, ' ', {
+            type: 'warning',
+            event: {
+              name: 'import-locale-differs',
+              data: {
+                moduleName,
+                exportPath,
+                content: {
+                  heading: req.t('aposImportExport:importWithCurrentLocaleHeading'),
+                  description: req.t('aposImportExport:importWithCurrentLocaleDescription', {
+                    docsLocale: differentDocsLocale,
+                    currentLocale: req.locale
+                  }),
+                  negativeLabel: 'apostrophe:no',
+                  affirmativeLabel: 'apostrophe:yes'
+                }
+              }
+            },
+            classes: [ 'apos-notification--hidden' ]
+          });
+
+          return;
+        }
+
+        // Re-write if user decided to continue the import (`overrideLocale` param sent)
+        // or if the site has only one locale configured
+        self.rewriteDocsWithCurrentLocale(req, docs);
```

Comment:
I suppose this method is called twice, another time with the confirmation to change locale?

Reply:
This method is called only during the 2nd call (when the user chooses Yes), see the early return above